### ISSUE TITLE
spring-configuration-metadata : must have primitive type property before F3 navigation works

### ISSUE THREAD

**Issue by space88man** [spring-projects/sts4#1030]
**Describe the bug**
When a `@ConfigurationProperties` does not have a field that is a primitive type it will not navigate with F3.

The root and child don't navigate with F3 and get `The operation is not applicable to the current selection`. The grandchild works as it has fields that are primitive type.

**To Reproduce**
```
app: # no primitive type field at all
  service1: # no primitive type field at all
    service: # this navigates
      host: 1.2.3.4
      port: 5000

package com.example.demo;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "app")
public class Settings {
    public static record Endpoint(String host, int port) {}
    public static record Service(Endpoint service) {}
    
    
    private Service service1;
    public Service getService1() {
        return service1;
    }
    public void setService1(Service service1) {
        this.service1 = service1;
    }
}

```
CAN'T navigate with F3: app, app.service1
CAN navigate with F3:  app.service1.service

**Sample**

[gh-sts4-sampleV2.zip](https://github.com/spring-projects/sts4/files/11331716/gh-sts4-sampleV2.zip)

**Comment by space88man** [comment]
It may be related to the fact that the children are records:

If I add a non-record field, then the app and app.value will navigate
```
@ConfigurationProperties(prefix = "app")
public class Settings {
    public static record Endpoint(String host, int port) {}
    public static record Service(Endpoint service) {}
    
    
    private Service service1;
    private Service service2;
    public Service getService1() {
        return service1;
    }
    public void setService1(Service service1) {
        this.service1 = service1;
    }
    public Service getService2() {
        return service2;
    }
    public void setService2(Service service2) {
        this.service2 = service2;
    }
    private int value;
    
    public int getValue() {
        return value;
    }
    public void setValue(int value) {
        this.value = value;
    }
}
```

**Comment by space88man** [comment]
Add a dummy field that is a primitive type makes F3 navigation work:

```
app2:
  value: true
  service1:
    value: true
    service:
      host: 1.2.3.4
      port: 5000
```

```
@ConfigurationProperties(prefix = "app2")
public class Settings2 {

    public static class Endpoint {
        private String host;
        private int port;
        public String getHost() {
            return host;
        }
        public void setHost(String host) {
            this.host = host;
        }
        public int getPort() {
            return port;
        }
        public void setPort(int port) {
            this.port = port;
        }
        
    }
    public static class Service {
        private Endpoint service;
        private boolean value;
        public Endpoint getService() {
            return service;
        }
        public void setService(Endpoint service) {
            this.service = service;
        }
        public boolean isValue() {
            return value;
        }
        public void setValue(boolean value) {
            this.value = value;
        }
    }


    private Service service1;
    private boolean value;

    public Service getService1() {
        return service1;
    }


    public void setService1(Service service1) {
        this.service1 = service1;
    }


    public boolean isValue() {
        return value;
    }


    public void setValue(boolean value) {
        this.value = value;
    }
}
```

**Comment by BoykoAlex** [comment]
Fixed with e9b75184984e133f22bc4c659b3b2bbd6c3ad2aa (and two commits above it)
Update from the nightly build. Find appropriate p2 repo here: https://dist.springsource.com/snapshot/STS4/nightly-distributions.html (see STS4 Distribution section at the top of the page)
The snapshot build should be available Apr 27 AM hours EST

**Comment by BoykoAlex** [comment]
@space88man Almost forgot... Thanks very much for taking the time to investigate the issue, providing great description and assembling a test project for this issue. Much appreciated by all of us on our end. We have included you test projects with use cases in our [unit tests](https://github.com/spring-projects/sts4/blob/main/headless-services/spring-boot-language-server/src/test/java/org/springframework/ide/vscode/boot/test/ApplicationYamlEditorTest.java#L1026)
